arviz/plots/densityplot.py

```diff
@@ -151,11 +151,14 @@ def plot_density(
 
         >>> az.plot_density([centered, non_centered], var_names=["mu"], bw=.9)
     """
-    if transform is not None:
-        data = transform(data)
+
     if not isinstance(data, (list, tuple)):
+        if transform is not None:
+            data = transform(data)
         datasets = [convert_to_dataset(data, group=group)]
     else:
+        if transform is not None:
+            data = [transform(datum) for datum in data]
```

Comment:
I think this won't work because here data is still inference data object, it should go somewhere below, once datasets has already been defined. This will also simplify the code because datasets is always a list, even if it has only one element.

Reply:
Ooh yeah, good catch! Plus, this will look more like the implementation in `plot_forest`